Remove checkout/edit and cancel actions

njohner opened the pull request:
* Removed the *checkout/edit* portal_type action.
* Removed the *checkout cancel* portal_type action and made a document-action.

**User checked out the document, manager can cancel the checkout**
![manager](https://user-images.githubusercontent.com/7374243/33328558-9d28d4f2-d45a-11e7-831d-300fde1ff962.png)

**Manager checked out document, user can't cancel**
![user](https://user-images.githubusercontent.com/7374243/33328561-9e7c0342-d45a-11e7-85a9-ffca2c840c20.png)


resolves #2905

Rotonen commented:
Users out there AFAIK primarily use the portal action bar for checking a document out instead of the file action buttons.

Can we get some user feedback from real users on this?

deiferni commented on opengever/core/upgrades/20171124163904_remove_edit_checkout_and_cancel_actions/types/opengever.document.document.xml:
```diff
@@ -0,0 +1,14 @@
+  </action>
+
+  <action
+      action_id="cancel_document_checkout"
```
does this actually work? i see some redundancy with `actions_remove_type_action`, i think only either `actions_remove_type_action` or the changes in this xml are necessary. could you double check?

_(the fact that we have an `actions_remove_type_action` method indicates that the removing actions via `xml` may not actually work.)_

njohner replied:
I first had only the `xml` remove and it did not work. So I added the `actions_remove_type_action` and then it worked. I'll test again with only the `actions_remove_type_action`, but without the `xml`...

deiferni replied:
👍  i strongly suspect the `xml` is not necessary 😉

njohner replied:
Yes you were right of course! xml remove does not seem to work, and the `actions_remove_type_action` alone is just fine

deiferni commented:
also if we merge this as is the documentation must be updated, see https://docs.onegovgever.ch/user-manual/dokumente/bearbeiten/#dokument-auschecken-und-bearbeiten

phgross commented:
@njohner some questions, but changes looks 👍 :

 - Are the revert icon already implemented in `plonetheme.teamraum`? if not please open a PR there as well and paste the link in to the PR description.
 - Is force checkin still possible for administrators and managers?

njohner commented:
I don't think the revert icon is in `plonetheme.teamraum`, so I'll create an issue there.
Force checkin still works for `Administrator` or `Verwalten` rights.

njohner commented:
That was a misunderstanding on my side, we don't need to do anything in `plonetheme.teamraum` here

deiferni commented:
@njohner i am looking at the open PRs. What is the state here? Is it ready for re-review or are you still working on this?

njohner commented:
@deiferni 
Just the documentation still needs to be updated. Should I do that?
For the rest it's ready

deiferni commented:
@njohner yes, please.

njohner commented:
@deiferni 
Ok I updated the docs, so this is ready for review. Thanks @siegy22 for local bumblebee installation!
Here are 2 snapshots from the user-manual showing the updated images and text.

**Only one button remaining for checkout:**
![screen shot 2018-01-15 at 16 48 00](https://user-images.githubusercontent.com/7374243/34950782-6b520ad6-fa14-11e7-8153-62d648abcaa2.png)

**Moved the cancel checkout button**
![screen shot 2018-01-15 at 16 47 16](https://user-images.githubusercontent.com/7374243/34950776-6a1548f4-fa14-11e7-8acb-d567e6c3e351.png)

njohner commented:
@deiferni @Rotonen 
Just had to adapt one more test (https://github.com/4teamwork/opengever.core/pull/3697/commits/62da6a1897e1ecd8b8efe98a17b030998680767d), which I think was ported by @Rotonen to integration layer since I first made this pull request. A one line change, but I'll wait for a thumbs up before merging.

Rotonen commented on opengever/document/locales/de/LC_MESSAGES/opengever.document.po:
```diff
@@ -23,6 +23,10 @@ msgstr "Mit Mailprogramm versenden"
 msgstr "Das Dokument kann momentan nicht bearbeitet werden, es ist gesperrt."
 
+#: ./opengever/document/browser/templates/macros.pt
+msgid "Cancel checkout"
```
Base already has a translation for this and there are similar ones within document as well - how do we end up with a new one here?

deiferni replied:
Good point. The translation has been moved to a page template. The previous implementation used actions, which have to be manually translated in the `plone` domain.
@njohner i think you could drop the manual translations for the checkout related stuff since the actions are dropped, too. You'll finde the `*.pot` file here, 
https://github.com/4teamwork/opengever.core/blob/master/opengever/base/locales/plone.pot, please also update the corresponding `*.po` files.

njohner commented:
@Rotonen @deiferni 
Ok I removed the translations from the `opengever/base/locales/plone.pot`. I rebased and as soon as the tests pass this should be ready.

njohner commented:
Oh noooo! It was obviously meant to be squashed 👎

njohner commented:
@deiferni Ok done! Sorry!
